core/src/main/java/org/wso2/testgrid/core/phase/TestPhase.java

```diff
@@ -146,22 +146,23 @@ private void runScenarioTests()
                 TestExecutor testExecutor = TestExecutorFactory.getTestExecutor(
                         TestEngine.valueOf(scenarioConfig.getTestType()));
 
-                Path scenarioDir = Paths.get(getTestPlan().getScenarioTestsRepository(), scenarioConfig.getName(),
-                        scenarioConfig.getFile());
-                if (scenarioDir !=  null) {
-                    Path parent = scenarioDir.getParent();
-                    Path file = scenarioDir.getFileName();
-                    if (parent == null) {
-                        parent = Paths.get("");
-                    }
-                    if (file == null) {
-                        file = Paths.get("test.sh");
+                if (scenarioConfig.getFile() != null) {
```

Comment:
If we can detect the non-existence of the file and then gracefully log that and exit, that would provide better user experience for the end-user. I don't think we should hide the error message, because this is an actual user error.